### PR TITLE
fix: citation generation

### PR DESCRIPTION
**Pull request by nayib-jose-gloria**
## Reason for Change

- #6345

## Changes

- updates dataset citation generator to use correct collections URL path

### CODE REVIEW

**Review by Bento007 — APPROVED** [review]

File: tests/unit/processing/test_process_validate.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def test_process_download_validate_success(self, mock_read_h5ad):
             citation_str = (
                 f"Publication: http://doi.org/12.2345 "
                 f"Dataset Version: http://domain/{dataset_version_id}.h5ad curated and distributed by "
-                f"CZ CELLxGENE Discover in Collection: http://collections/{collection.collection_id.id}"
+                f"CZ CELLxGENE Discover in Collection: http://domain/collections/{collection.collection_id.id}"
```
Comment (danieljhegeman) [inline tests/unit/processing/test_process_validate.py:54]:
```suggestion
                f"CZ CELLxGENE Discover in Collection: https://domain/collections/{collection.collection_id.id}"
```

File: tests/unit/processing/test_process_validate.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def test_populate_dataset_citation__no_publication_doi(self, mock_read_h5ad):
         pdv.populate_dataset_citation(collection.version_id, dataset_version_id, "")
         citation_str = (
             f"Dataset Version: http://domain/{dataset_version_id}.h5ad curated and distributed by "
-            f"CZ CELLxGENE Discover in Collection: http://collections/{collection.collection_id.id}"
+            f"CZ CELLxGENE Discover in Collection: http://domain/collections/{collection.collection_id.id}"
```
Comment (danieljhegeman) [inline tests/unit/processing/test_process_validate.py:82]:
```suggestion
            f"CZ CELLxGENE Discover in Collection: https://domain/collections/{collection.collection_id.id}"
```

File: tests/unit/processing/test_process_validate.py (hunk 1 above)
Comment (danieljhegeman) [reply tests/unit/processing/test_process_validate.py:54]: nit but just for clarity/consistency. Same below

**Review by danieljhegeman — APPROVED** [review]
Looks good but see suggestion for test url: use correct protocol `https`

File: tests/unit/backend/layers/common/base_test.py [hunk 3]
```diff
@@ -81,7 +81,7 @@ def setUp(self):
         os.environ.setdefault("APP_NAME", "corpora-api")
         config = {
             "upload_max_file_size_gb": 30,
-            "collections_base_url": "http://collections",
+            "collections_base_url": "http://domain",
```
Comment (danieljhegeman) [inline tests/unit/backend/layers/common/base_test.py:84]:
```suggestion
            "collections_base_url": "https://domain",
```

**Pull request merged**